### PR TITLE
Issue #1329

### PR DESCRIPTION
**Pull request by gregw**
For #1329 

Allow extended classes to access the LogEntry building, so that
additional information may be encoded in each log entry (eg traceid).

Use addLabel rather than setLabels as it avoids the creation and copy
of a HashMap per log entry.

Signed-off-by: Greg Wilkins <gregw@webtide.com>

### CODE REVIEW

File: google-cloud-logging/src/main/java/com/google/cloud/logging/LoggingHandler.java [hunk 1]
```diff
@@ -303,15 +303,17 @@ private LogEntry entryFor(LogRecord record) {
...
-        .setSeverity(severityFor(level))
-        .build();
+    LogEntry.Builder builder = LogEntry.newBuilder(Payload.StringPayload.of(payload))
+        .addLabel("levelName", level.getName())
+        .addLabel("levelValue", String.valueOf(level.intValue()))
+        .setSeverity(severityFor(level));
+    return buildEntryFor(record, builder);
   }
 
+  protected LogEntry buildEntryFor(LogRecord record, LogEntry.Builder builder) {
```
Comment (mziccard) [inline google-cloud-logging/src/main/java/com/google/cloud/logging/LoggingHandler.java]: Isn't adding this method enough for your needs? Why making also `entryFor` protected?

Comment (gregw) [inline google-cloud-logging/src/main/java/com/google/cloud/logging/LoggingHandler.java]: Correct buildEntryFor is sufficient for our needs.
I made `entryFor` protected as my first approach was going to be to replace that method.  I then left it protected as I thought others might like to extend it.   But we don't need it.   

Shall I remove it from this PR?

**Comment by meltsufin** [comment]
@mziccard Work on the jetty-runtime is blocking on this. Is anything else needed to approve this for merging?

**Comment by mziccard** [comment]
@meltsufin I am no longer involved in this project, please refer to @garrettjonesgoogle for help

File: google-cloud-logging/src/main/java/com/google/cloud/logging/LoggingHandler.java (hunk 1 above)
Comment (garrettjonesgoogle) [inline google-cloud-logging/src/main/java/com/google/cloud/logging/LoggingHandler.java]: Hi @gregw , I find this signature to be a little unusual. Would you be able to give more details about why it needs to look this way in particular?

Comment (garrettjonesgoogle) [inline google-cloud-logging/src/main/java/com/google/cloud/logging/LoggingHandler.java]: I think the following would be a better abstraction:

```
protected void enhanceLogEntry(LogEntry.Builder builder, LogRecord record) {
  // no-op in this class
}
```

and do the `.build()` call in `entryFor`.

**Comment by garrettjonesgoogle** [comment]
LGTM if Travis passes

**Pull request merged**